utils/spark_utils.py

```diff
@@ -759,3 +740,64 @@ def get_standard_concept_ids(spark, concept_ids):
             WHERE ca.concept_id_1 IN ({concept_ids})
         """.format(concept_ids=','.join([str(c) for c in concept_ids])))
     return standard_concept_ids
+
+
+def link_events_with_visits(patient_event):
```

Comment:
If two visits occur on the same day with different visit types (ED + Inpatient), the two last functions may not consistently take the information from the same visit.

Reply:
Thanks this is a case I hadn't considered. If we have two visits on the same day and an orphan event, should we try to link it to a visit? We don't have enough information I think to find out which would be the correct visit.

Reply:
@egillax This is probably a rare event, but this is more likely to happen if we specify a large gap period other than the same day. My point is that it doesn't matter to which visit we link the orphan events, however, it's important to link them consistently.

Reply:
@ChaoPang I agree about consistency. 

I'll look into joining directly to the `visit_occurrence`

Reply:
@ChaoPang I've been struggling a bit with your suggestion to join directly to the `visit_occurrence` to get the ids. I'm very new to pyspark. Could you explain it a bit maybe?